### ISSUE TITLE
Safe to ignore warning when exporting a react-redux connected component?

### ISSUE THREAD

**Issue by joshgreco** [ArnaudBarre/eslint-plugin-react-refresh#51]
This might be a really dumb question that belies my lack of understanding, but I'm getting the _react-refresh/only-export-components_ warning when exporting a [react-redux](https://react-redux.js.org/) connected component like so:

`export default connect(mapStateToProps, mapDispatchToProps)(MyComponent)`

I'm wondering if its "safe" to ignore this warning? My understanding is that [connect()](https://react-redux.js.org/api/connect#connect-returns) returns a wrapper function which in turn returns a connected component. Is this technically an anonymous component?

The warning goes away if I change things to:

```
const ConnectedMyComponent = connect(mapStateToProps, mapDispatchToProps)(MyComponent);
export default ConnectedMyComponent;
```

But is that really doing anything besides circumventing the warning?

Thank you, and apologies if this is incredibly stupid of me to ask.

**Comment by ArnaudBarre** [comment]
That not stupid to ask. The rule can't be sure that this connect function returns a component and that this component will be correctly named (important for stack trace). I will verify that the component is correctly named and add an exception for this connect function
